### ISSUE TITLE
Add import switching from `typing_extensions` to `typing` when reaches requisite version.

### ISSUE THREAD

**Issue by nstarman** [astral-sh/ruff#7285]
`typing_extensions` backports a lot of code from `typing`. It would be great if `ruff` could switch the import from `typing_extensions` to `typing` when ruff's target version matches (or exceeds) the requisite python version.

E.g.
```python
from typing_extensions import Annotated
```
Becomes 
```
from typing import Annotated
```
when ruff's target version is 3.9 or greater.

This feels like a pyupgrade code?

**Comment by charliermarsh** [comment]
I think this does exist as `UP035`: https://beta.ruff.rs/docs/rules/deprecated-import/

I tested it locally with `check foo.py --fix -n --select UP035 --target-version py39` and it did rewrite that import for me. Any luck?

**Comment by nstarman** [comment]
Yes, thanks! I was using an old test repo with target version set to py3.8. Updating that made everything work. Sorry for the noise!
